src/core/Akka/IO/SocketEventArgsPool.cs

```diff
@@ -21,48 +22,81 @@ public interface ISocketEventArgsPool
     {
         SocketAsyncEventArgs Acquire(IActorRef actor);
         void Release(SocketAsyncEventArgs e);
+        
+        BufferPoolInfo BufferPoolInfo { get; }
     }
 
     internal class PreallocatedSocketEventAgrsPool : ISocketEventArgsPool
     {
+        private readonly IBufferPool _bufferPool;
         private readonly EventHandler<SocketAsyncEventArgs> _onComplete;
-        private readonly ConcurrentStack<SocketAsyncEventArgs> _pool = new ConcurrentStack<SocketAsyncEventArgs>();
+        private readonly ConcurrentQueue<SocketAsyncEventArgs> _pool = new ConcurrentQueue<SocketAsyncEventArgs>();
```

Comment:
Changed to make sure that the content of the pool are rotated evenly, instead of only using the top ones. Minimizes the problem where SocketAsyncEventArgs could not be reused because it is locked by the previous socket.
To be honest, this whole class is a bad idea because SocketAsyncEventArgs are only supposed to be recycled if they're used by a single socket.